asset-transfer-basic/application-gateway-typescript/src/app.ts

```diff
@@ -205,3 +223,16 @@ async function updateNonExistentAsset(contract: Contract): Promise<void>{
         console.log('*** Successfully caught the error: \n', error);
     }
 }
+
+/**
+ * envOrDefault() will return the value of an environment variable, or a default value if the variable is undefined.
+ */
+function envOrDefault(key: string, defaultValue: string): string {
+
+    const value = process.env[key];
+    if (! value) {
+        return defaultValue;
+    }
+
+    return value;
+}
```

Comment:
Logically this is fine. I'd probably expect a more concise form for Type/JavaScript code, as the behaviour of the `||` operator is well defined and it is frequently used:
```suggestion
function envOrDefault(key: string, defaultValue: string): string {
    return process.env[key] || defaultValue;
}
```

In general I like the separate named function to document the intent. The only downside with using `envOrDefault(key, defaultValue)` instead if directly using `process.env[key] || defaultValue` is that the default value is always calculated in order to invoke the function, whereas the inline approach avoids that calculation if the environment variable is present. For this example, that isn't really a concern though, so whichever approach you prefer.